mullvad-relay-selector/tests/relay_selector.rs

```diff
@@ -0,0 +1,1048 @@
+/// still return a relay of the correct tunnel type (OpenVPN).
+#[test]
+fn test_prefer_openvpn_if_location_supports_it() {
+    let relay_selector = default_relay_selector();
+    let query = RelayQueryBuilder::new()
+        .location(GeographicLocationConstraint::hostname(
+            "se",
+            "got",
+            "se-got-001",
+        ))
+        .build();
+
+    for _ in 0..RETRY_ORDER.len() {
+        let relay = relay_selector.get_relay_by_query(query.clone()).unwrap();
+        let tunnel_typ = tunnel_type(&unwrap_relay(relay));
+        assert_eq!(tunnel_typ, TunnelType::OpenVpn);
+    }
+}
+
+/// Assert that the relay selector does *not* return a multihop configuration where the exit and entry relay are
+/// the same, even if the constraints would allow for it. Also verify that the relay selector is smart enough to
+/// pick either the entry or exit relay first depending on which one ends up yielding a valid configuration.
+#[test]
+fn test_wireguard_entry() {
+    let relay_selector = default_relay_selector();
+    let specific_hostname = "se10-wireguard";
+    let specific_location = GeographicLocationConstraint::hostname("se", "got", specific_hostname);
+    let general_location = GeographicLocationConstraint::city("se", "got");
+
+    for _ in 0..100 {
+        let query = RelayQueryBuilder::new()
+            .wireguard()
+            .location(general_location.clone())
+            .multihop()
+            .entry(specific_location.clone())
+            .build();
+
+        // The exit relay must not equal the entry relay
+        let relay = relay_selector.get_relay_by_query(query).unwrap();
+        match relay {
+            GetRelay::Wireguard {
+                inner: WireguardConfig::Multihop { exit, entry },
+                ..
+            } => {
+                assert_ne!(exit.hostname, entry.hostname);
+            }
+            wrong_relay => panic!(
+            "Relay selector should have picked a Wireguard relay, instead chose {wrong_relay:?}"
+        ),
+        }
+    }
+
+    for _ in 0..100 {
+        let query = RelayQueryBuilder::new()
+            .wireguard()
+            .location(specific_location.clone())
+            .multihop()
+            .entry(general_location.clone())
+            .build();
+
+        let relay = relay_selector.get_relay_by_query(query).unwrap();
+        match relay {
+            GetRelay::Wireguard {
+                inner: WireguardConfig::Multihop { exit, entry },
+                ..
+            } => {
+                assert_ne!(exit.hostname, entry.hostname);
+                assert_ne!(exit.ipv4_addr_in, entry.ipv4_addr_in);
+                assert_eq!(exit.hostname, specific_hostname)
+            }
+            wrong_relay => panic!(
+            "Relay selector should have picked a Wireguard relay, instead chose {wrong_relay:?}"
+        ),
+        }
+    }
```

Comment:
This part depends on the relay list containing exactly two wireguard relay in gothenburg, right? It would probably be a little unclear for someone how changed the `RELAYS` list why the test suddenly fails. Maybe you should split this part into a separate test and instanciate the relay selector with those just those two relays specifically.

Reply:
Fixed, thanks!